ios/Classes/AirshipInboxMessageView.swift

```diff
@@ -91,6 +98,40 @@ class AirshipInboxMessageView : NSObject, FlutterPlatformView, UANativeBridgeDel
     }
 
     func close() {
-        // In the future we should send a close event to the dart layer
+        AirshipEventManager.shared.notify(closeEvent)
+    }
+    
+    func webView(_ webView: WKWebView, decidePolicyFor navigationResponse: WKNavigationResponse,
+                 decisionHandler: @escaping (WKNavigationResponsePolicy) -> Void) {
+
+        if let response = navigationResponse.response as? HTTPURLResponse {
+            if (response.statusCode >= 400 && response.statusCode <= 599) {
+                decisionHandler(.cancel)
+                if (response.statusCode == 410) {
+                    webviewResult(FlutterError(code:"MessageLoadFailed",
+                           message:"Unable to load message",
+                           details:"Message not available"))
+                } else {
+                    webviewResult(FlutterError(code:"MessageLoadFailed",
+                           message:"Unable to load message",
+                           details:"Message load failed"))
+                }
+            }
+        }
+        decisionHandler(.allow)
+    }
+    
+    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
+        AirshipEventManager.shared.notify(loadFinishedEvent)
+    }
+    
+    func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
+        webviewResult(FlutterError(code:"MessageLoadFailed",
+        message:"Unable to load message",
+        details:error.localizedDescription))
+    }
+    
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        webView.navigationDelegate?.webView?(webView, didFail: navigation, withError: error)
```

Comment:
And do you need to emit a `loadFinishedEvent` here? (sorry for all of the duplicate comments)